guides/10_Video_Processor.md

```diff
@@ -0,0 +1,115 @@
+    processors: VideoFrameProcessor[],
+    browserBehavior: DefaultBrowserBehavior
+  )
+}
+```
+
+#### Construction and Starting Video Processing
+
+The construction of the `DefaultVideoTransformDevice` will not start the camera or start processing. The method `meetingSession.audioVideo.chooseVideoInputDevice` is needed to be called. Device controller will use the inner `Device` to acquire the source `MediaStream` and start the processing pipeline at the same frame rate as the source `MediaStream`.
+The parameters to `chooseVideoInputQuality` will be respected when device controller acquires the source `MediaStream`. 
+After video input is chosen, `meetingSession.audioVideo.startLocalVideoTile` can be called to start streaming video.
+
+
+#### Switching Device on VideoTransformDevice
+
+To switch the inner `Device` on `DefaultVideoTransformDevice`, call `DefaultVideoTransformDevice.chooseNewInnerDevice` with a new `Device`.
+`DefaultVideoTransformDevice.chooseNewInnerDevice` returns a new `DefaultVideoTransformDevice` but preserves the state of `VideoFrameProcessor`s. Then call `meetingSession.audioVideo.chooseVideoInputDevice` with the new transform device. 
+
+
+#### Stopping VideoTransformDevice
+
+To stop video processing for the chosen `DefaultVideoTransformDevice`, call `meetingSession.audioVideo.chooseVideoInputDevice` with different `DefaultVideoTransformDevice` or `null` to stop using previous `DefaultVideoTransformDevice`. Or `meetingSession.audioVideo.stopLocalVideoTile` can be used to stop the streaming.
+
+After the call, the inner `Device` will be released by device controller unless the inner `Device` is a `MediaStream` provided by users and it is the responsibility of users to handle the lifecycle. 
+
+After `DefaultVideoTransformDevice` is no longer used by device controller, call `DefaultVideoTransformDevice.stop` to release the `VideoProcessor`s and underlying pipeline. After `stop` is called, users must discard the `DefaultVideoTransformDevice`.`DefaultVideoTransformDevice.stop` is necessary to release all the resources.
+
+
+#### Receiving lifecycle notifications with observer
+
+To receive notifications of lifecycle events, `DefaultVideoTransformDeviceObserver` can be added via `addObserver` method. 
+The full list of the callbacks:
+
+1. `processingDidStart`
+
+`processingDidStart` will be called when video processing starts.
+
+2. `processingDidFailToStart`
+
+`processingDidFailToStart` will be called when video processing could not start due to runtime errors. In this case, developers are expected to call `chooseVideoInputDevice` again with a valid `VideoInputDevice` to continue video sending. 
+
+3. `processingDidStop`
+
+`processingDidStop` will be called when video processing is stopped **expectedly**.
+
+4. `processingLatencyTooHigh(latencyMs: number)` 
+
+`processingLatencyTooHigh` will be called when the execution of processors slows the frame rate down by at least half.
+
+
+### VideoFrameProcessorPipeline
```

Comment:
Do we need it in the guide if builders do not need to handle this?

Reply:
`### VideoFrameBuffer and VideoFrameProcessor` can be kept tho